typescript/sdk/src/token/schemas.ts

```diff
@@ -71,7 +95,72 @@ export const WarpRouteDeployConfigSchema = z
     const entries = Object.entries(configMap);
     return (
       entries.some(
-        ([_, config]) => isCollateralConfig(config) || isNativeConfig(config),
+        ([_, config]) =>
+          isCollateralConfig(config) ||
+          isCollateralRebaseConfig(config) ||
+          isNativeConfig(config),
       ) || entries.every(([_, config]) => isTokenMetadata(config))
     );
-  }, `Config must include Native or Collateral OR all synthetics must define token metadata`);
+  }, WarpRouteDeployConfigSchemaErrors.NO_SYNTHETIC_ONLY)
+  .transform((warpRouteDeployConfig, ctx) => {
+    const hasCollateralRebase = Object.entries(warpRouteDeployConfig).find(
+      ([_, config]) => isCollateralRebaseConfig(config),
+    );
+    if (!hasCollateralRebase) return warpRouteDeployConfig;
```

Comment:
zod `.refine` may be slightly more apt

Reply:
i prefer transform here because it's actually transforming. Think i had some trouble using refine too